service/resource/certificate/update.go

```diff
@@ -79,6 +79,9 @@ func (r *Resource) ProcessUpdateState(ctx context.Context, obj, updateState inte
 		}
 	}
 
+	r.logger.Log("debug", "certificates have been updated, requesting reload")
+	r.prometheusReloader.RequestReload()
```

Comment:
I don't understand the fix here. This requests the reload after the first object given by the informer will be updated. I understand that we then have written certs for one guest cluster. There might be 2 others we did not write certs for yet. So this will break again as soon as there is more than one guest cluster to scrape. I feel like I am missing something.

Reply:
We process all of the certificates, possibly requesting reloads. When the configmap is updated (even if there is no change), if a reload has been requested, we fire a reload.

Reply:
What is the custom object then provided by the informer when the scope of the resource is actually all resources?

Reply:
I don't follow

Reply:
What does the informer give out?

Reply:
Services!

Reply:
We discussed. There will be refactoring and such.